Management API Voting Basics

maael opened the pull request:
Adds the basics for the management api, starting with some voting apis, create/get/list. It also adds a list of votes to the home page.

maael committed:
:ballot_box_with_check: Add basics for management api for votes

maael committed:
:wrench: Fix the migration names for management

maael committed:
:chart_with_upwards_trend: Add vote list to home page

MetalMichael commented on EvotoApi/Areas/Management/Models/Request/CreateManaVote.cs:
```diff
@@ -0,0 +1,47 @@
+    {
+        [DataMember(Name = "orgId")]
+        [Required]
+        public int OrgId { get; }
```
These need setters, as mentioned in Slack PM

maael replied:
Added in https://github.com/evoto-tech/EvotoApi/pull/12/commits/e2af60ca0bae4aab7d4da01f303273aef5806bac.

maael committed:
:squirrel: Add private setters for create vote model

maael committed:
:clock1: Add bootstrap-datetimepicker and dependencies

maael committed:
:heavy_plus_sign: Add new vote creation page

- Hooked up to api
- ISSUE: bad request when created

MetalMichael commented on Management.Database/Models/ManaDbVote.cs:
```diff
@@ -0,0 +1,59 @@
+        {
+            return new ManaVote
+            {
+                Id = (int) Id,
```
Why are these casted?

maael replied:
Not overly sure. I feel like this might be c&p residual, https://github.com/evoto-tech/EvotoApi/blob/e264b2ca1aa1d0b05824f8b469eca5172d642eaa/Registrar.Database/Models/RegiDbUser.cs#L33. Monkey see monkey do 🙈.

MetalMichael replied:
Fair. I think that's left over from when I was using uint's in the database models. Is removed in my local version, probably not been merged yet.

maael replied:
Fixed in https://github.com/evoto-tech/EvotoApi/pull/12/commits/cfda12d7a5cc89aa6e88fd131beb3b4773c71ddd.

MetalMichael commented on Management.Database/Stores/ManaSqlVoteStore.cs:
```diff
@@ -0,0 +1,104 @@
+                    if (!result.Any())
+                        throw new RecordNotFoundException();
+
+                    return result.Select((v) => new ManaDbVote(v)).Select((v) => v.ToVote());
```
no need to iterate through twice. Just `Select(v => new ManaDbVote(v).ToVote());`

maael replied:
Fixed in https://github.com/evoto-tech/EvotoApi/pull/12/commits/951b803fde3d686d519b0a28fc5e0f07d2f95578.

maael committed:
:arrows_clockwise: Reduce need to loop over votes from db

Addresses https://git.io/vMHwk.

maael committed:
:fishing_pole_and_fish: Remove unnecessary casting

Addresses https://git.io/vMHw8.

maael commented on Management.Database/Stores/ManaSqlVoteStore.cs:
```diff
@@ -0,0 +1,104 @@
+                    if (!result.Any())
+                        throw new RecordNotFoundException();
+
+                    return result.Select((v) => new ManaDbVote(v)).Select((v) => v.ToVote());
```
Fixed in https://github.com/evoto-tech/EvotoApi/pull/12/commits/cfda12d7a5cc89aa6e88fd131beb3b4773c71ddd.

maael committed:
:office: Remove organisation concept from votes

As discussed in slack pm.

maael committed:
:wrench: Fix js linting, add optional modal content to warning modal

- Also add in the confirm button to the modal after it was accidentally
  taken out.

maael committed:
:100: Finish removing organisations from votes

maael committed:
:wrench: Fix creating a vote via management api

maael committed:
:interrobang: Add errors to new vote page

maael committed:
:arrow_right: Fix login redirect

maael committed:
:globe_with_meridians: Generalise NewVote so it can take an existing vote

This is prelim work for editing a vote.

maael committed:
:pencil: FE work for editing a vote

TODO: Backend PUT route needs implementing.

maael committed:
:wrench: Fix js lint issues

maael committed:
:pencil: Finish editing votes functionality

:wrench: Also fix expiryDate when creating a vote.

maael committed:
:evergreen_tree: Remove some leftover logging

maael committed:
:evergreen_tree: Remove some leftover logging

maael committed:
:x: Add ability to delete unpublished votes

TODO: Add server side validation to only allow unpublished votes to be
deleted.

maael committed:
:x: Add serverside validation to only allow deleting unpublished votes
